lib/queryBuilder/graphUpserter/UpsertNode.js

```diff
@@ -327,6 +348,22 @@ function hasChanges(currentModel, upsertModel) {
   };
 }
 
+function hasRelationsInUpsertModel(upsertModel) {
+  const relationArray = upsertModel.constructor.getRelationArray();
+  const changingProps = [];
+
+  for (let i = 0, l = relationArray.length; i < l; ++i) {
+    const relation = relationArray[i];
+    const upsertRelated = upsertModel[relation.name];
+
+    if (upsertRelated) {
+      changingProps.push(upsertRelated);
```

Comment:
The array is not needed. You could just return true here and remove the array right?